src/include/duckdb/main/secret/secret.hpp

```diff
@@ -100,6 +100,10 @@ class BaseSecret {
 	//! Serialize this secret
 	virtual void Serialize(Serializer &serializer) const;
 
+	virtual unique_ptr<const BaseSecret> Clone() const {
+		return make_uniq<BaseSecret>(*this);
```

Comment:
Can add a check `D_ASSERT(typeid(BaseSecret) == typeid(*this))` <-- this will assert that Clone is not being overridden by a subclass by mistake.

Reply:
Thanks for the review. I've added this assertion.